source/common/signal_handler.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <thread>
+#include <vector>
+
+#include "external/envoy/source/common/common/logger.h"
+
+namespace Nighthawk {
+
+using SignalCallback = std::function<void()>;
+
+class SignalHandler final : public Envoy::Logger::Loggable<Envoy::Logger::Id::main> {
+public:
+  SignalHandler(const SignalCallback& signal_callback);
+  SignalHandler(SignalHandler const&) = delete;
```

Comment:
Can we add a single comment above these two to indicate the intention, something like:

// Not copyable or movable.